dynawo/sources/Models/CPP/ModelLoadRestorativeWithLimits/DYNModelLoadRestorativeWithLimits.cpp

```diff
@@ -0,0 +1,577 @@
+    parameters.push_back(ParameterModeler("load_UDeadBandPu", VAR_TYPE_DOUBLE, EXTERNAL_PARAMETER));
+  }
+
+  void
+  ModelLoadRestorativeWithLimits::setSubModelParameters() {
+    try {
+    u0_ = findParameterDynamic("load_U0Pu").getValue<double>();
+    Tf_ = findParameterDynamic("load_tFilter").getValue<double>();
+    P0_ = findParameterDynamic("load_P0Pu").getValue<double>();
+    Q0_ = findParameterDynamic("load_Q0Pu").getValue<double>();
+    alpha_ = findParameterDynamic("load_Alpha").getValue<double>();
+    beta_ = findParameterDynamic("load_Beta").getValue<double>();
+    angleO_ = findParameterDynamic("load_UPhase0").getValue<double>();
+    UMinPu_ = findParameterDynamic("load_UMin0Pu").getValue<double>();
+    UMaxPu_ = findParameterDynamic("load_UMax0Pu").getValue<double>();
+    double UDeadBandPu = findParameterDynamic("load_UDeadBandPu").getValue<double>();
+    if ((u0_ - UDeadBandPu) < UMinPu_)
+      UMinPu_ = u0_ - UDeadBandPu;
+    if ((u0_ + UDeadBandPu) > UMaxPu_)
+      UMaxPu_ = u0_ + UDeadBandPu;
+    } catch (const DYN::Error& e) {
+    Trace::error() << e.what() << Trace::endline;
+    throw DYNError(Error::MODELER, NetworkParameterNotFoundFor, staticId());
+    }
+  }
+
+  void
+  ModelLoadRestorativeWithLimits::defineVariables(std::vector<boost::shared_ptr<Variable> >& variables) {
+    variables.push_back(VariableNativeFactory::createState("UfRaw_value", CONTINUOUS));
+    variables.push_back(VariableNativeFactory::createState("Uf_value", CONTINUOUS));
+    variables.push_back(VariableNativeFactory::createState("Ur_value", CONTINUOUS));
+    variables.push_back(VariableNativeFactory::createState("Ui_value", CONTINUOUS));
+    variables.push_back(VariableNativeFactory::createState("Ir_value", FLOW));
+    variables.push_back(VariableNativeFactory::createState("Ii_value", FLOW));
+    variables.push_back(VariableNativeFactory::createState("P_value", CONTINUOUS));
+    variables.push_back(VariableNativeFactory::createState("Q_value", CONTINUOUS));
+    variables.push_back(VariableNativeFactory::createCalculated("state_value", CONTINUOUS));
+    variables.push_back(VariableNativeFactory::createState("switchOff1_value", BOOLEAN));
+    variables.push_back(VariableNativeFactory::createState("switchOff2_value", BOOLEAN));
+  }
+
+  void
+  ModelLoadRestorativeWithLimits::init(const double& /*t0*/) {
+    unsigned int localIndex = 0;
+    UfRawYNum_ = localIndex;
+    ++localIndex;
+    UfYNum_ = localIndex;
+    ++localIndex;
+    UrYNum_ = localIndex;
+    ++localIndex;
+    UiYNum_ = localIndex;
+    ++localIndex;
+    IrYNum_ = localIndex;
+    ++localIndex;
+    IiYNum_ = localIndex;
+    ++localIndex;
+    PYNum_ = localIndex;
+    ++localIndex;
+    QYNum_ = localIndex;
+  }
+
+  void
+  ModelLoadRestorativeWithLimits::collectSilentZ(BitMask* /*silentZTable*/) {
+    /* not need */
+  }
+
+  void
+  ModelLoadRestorativeWithLimits::getSize() {
+    if (getIsInitProcess()) {
```

Comment:
getIsInitProcess() will never be true for a classical CPP model so you can get rid of the if